File: 0101-0200/0172.Factorial_Trailing_Zeroes.cpp
//0ms





class Solution {
public:
    int trailingZeroes(int n) {
        int ans = 0;
        int div = 5;

        while(n>=div)
        {
            ans+= (n/div);
            div*=5;
        }

        return ans;
    }
};






//0ms






class Solution {
public:
    int trailingZeroes(int n) {
        int c=0;
        for(int i=5;i<=n;i=i*5){
            c+=n/i;
        }
        return c;
    }
};

File: 0101-0200/0172.Factorial_Trailing_Zeroes.py
#0ms






class Solution:
    def trailingZeroes(self, n: int) -> int:
        res = 0
        div = 1
        while div < n:
            div *= 5
            res += n // div
        return res






//0ms






class Solution:
    def trailingZeroes(self, n: int) -> int:
        res = 0
        while n > 0:
            n //= 5
            res += n
        return res
